velox/vector/tests/EnsureWritableVectorTest.cpp

```diff
@@ -817,22 +817,51 @@ TEST_F(EnsureWritableVectorTest, allNullMap) {
 }
 
 TEST_F(EnsureWritableVectorTest, booleanFlatVector) {
-  VectorPtr vector =
-      makeFlatVector<bool>(100, [](auto /*row*/) { return true; });
-
-  // Make sure vector::values_ buffer is not uniquely referenced so that the
-  // branch in the FlatVector::ensureWritable() that copy old values to new
-  // buffer is executed.
-  auto another = vector->asFlatVector<bool>()->values();
-
-  SelectivityVector rows{200, false};
-  rows.setValidRange(16, 32, true);
-  rows.updateBounds();
-
-  auto vectorPtr = vector.get();
-  ASSERT_NO_THROW(BaseVector::ensureWritable(rows, BOOLEAN(), pool(), vector));
-  ASSERT_EQ(vectorPtr, vector.get());
-  ASSERT_NE(another->as<void>(), vector->valuesAsVoid());
+  auto testEnsureWritableBoolean =
+      [this](VectorPtr& vector, SelectivityVector& rows, BufferPtr& origin) {
+        auto vectorPtr = vector.get();
+        ASSERT_NO_THROW(
+            BaseVector::ensureWritable(rows, BOOLEAN(), pool(), vector));
+        ASSERT_EQ(vectorPtr, vector.get());
+
+        // Make sure vector::values_ buffer is not uniquely referenced so that
```

Comment:
This comment is no longer attached to the "right" code. It seems like you can remove 'origin' parameter and add a local variable here (another) like i the original code.